apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPoolImpl.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.V1.Logging;
+
+namespace Google.Cloud.Spanner.V1
+{
+    internal class SessionPoolImpl : IComparable
```

Comment:
Use the generic `IComparable<T>` unless you really need the non-generic one. Or potentially have a separate `IComparer<SessionPoolImpl>` to make it clear how it's comparing the session pools...